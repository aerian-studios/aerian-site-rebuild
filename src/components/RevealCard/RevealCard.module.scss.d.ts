/* tslint:disable */
// This file is automatically generated from your CSS. Any edits will be overwritten.
export const revealCard: string;
export const flip: string;
export const visible: string;
export const seen: string;
export const revealCardWrapper: string;
export const revealCardContent: string;
export const clientPromoLogo: string;
export const cardAlignment: string;
