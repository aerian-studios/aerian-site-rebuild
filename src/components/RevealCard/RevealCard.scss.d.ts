/* tslint:disable */
// This file is automatically generated from your CSS. Any edits will be overwritten.
export const component: string;
export const bgImg: string;
export const anchor: string;
export const clientLogo: string;
export const overlay: string;
export const button: string;
