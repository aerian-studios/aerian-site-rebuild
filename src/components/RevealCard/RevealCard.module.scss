@import "../../scss/variables";
.revealCard {
    width: 100%;
    height: 100%;
    position: relative;

    .flip {
        transition: transform 0.5s cubic-bezier(0.455, 0.03, 0.515, 0.955);
        transform-origin: top center;
        transform: translate3d(0, 0, 0) rotateX(90deg) rotateZ(-10deg);
        transition-delay: 0.2s;
        opacity: 0.1;
        will-change: transform;
        backface-visibility: hidden;
    }
    &.visible .flip,
    &:first-child .flip,
    &.seen .flip {
        transform: rotateX(0deg) rotateZ(0deg);
        opacity: 1;
        will-change: auto;
    }
}

.revealCardWrapper {
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 1fr;
    color: $whiteish;

    > * {
        position: absolute;
        top: 0;
        left: 0;
    }

    &:hover,
    &:focus {
        .revealCardContent {
            opacity: 1;
        }

        .clientPromoLogo {
            transform: translateY(-110%);
        }
    }
}

.cardAlignment {
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    grid-column: 1/2;
    grid-row: 1/2;
}

.clientPromoLogo {
    composes: cardAlignment;
    max-width: 79%;
    max-height: 25%;
    align-self: center;
    justify-self: center;
    text-align: center;
    transition: transform 0.5s ease 0.2s;

    // override responsive image styles!!
    picture,
    img {
        height: initial !important;
        max-height: 100% !important;
        max-width: 100% !important;
        width: initial !important;
        transform: translate(-50%, -50%);
        left: 50% !important;
        top: 50% !important;
    }
}

.revealCardContent {
    composes: cardAlignment;
    display: flex;
    flex-flow: column wrap;
    justify-content: center;
    align-items: center;
    background-color: rgba(15, 8, 4, 0.45);
    opacity: 0;
    transition: opacity 0.8s ease 0.2s;
    z-index: 2;
    padding: 12.5%;
    color: $whiteish;
}
