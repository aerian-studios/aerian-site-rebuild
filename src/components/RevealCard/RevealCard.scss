.component {
    color: inherit;
    position: relative;
}
.bgImg {
    position: absolute;
    height: 100%;
    width: 100%;
    top: 0;
    left: 0;

    > * {
        object-fit: cover;
    }
}

.anchor {
    display: flex;
    height: 100%;
    width: 100%;

    overflow: hidden;
    justify-content: center;
    align-items: center;
}

.clientLogo {
    flex: 1;
}

.overlay {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    display: flex;
}

.overlay:hover {
    background-color: rgba(0, 0, 0, 0.3);
    transition: all 0.75s ease;
}
.button {
    composes: button from "../../scss/base.buttons.scss";
}
