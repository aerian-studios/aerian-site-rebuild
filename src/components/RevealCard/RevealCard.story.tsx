
import * as React from "react";

import { withInfo } from "@storybook/addon-info";
import { storiesOf } from "@storybook/react";

import { RevealCard } from "./index";


storiesOf("RevealCard", module).add(
    "Default", 
    withInfo({ inline: true })(() => (
    <RevealCard  className="myClass" />
)));
