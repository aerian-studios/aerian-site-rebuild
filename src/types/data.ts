import { Stats } from "fs";
import { ParsedPath } from "path";

export type ImageField = string | ImageSharp;
export type FileField = string | Partial<FileNode>;

// Images

export interface GalleryImage {
    alt?: string;
    image: ImageField;
}

export interface FileNode extends ParsedPath, Stats {
    publicURL: string;
    id: string;
    children: any[];
    sourceInstanceName: string;
    absolutePath: string;
    relativePath: string;
    relativeDirectory: string;
    extension: string;
    size: number;
    prettySize: string;

    modifiedTime: string;
    accessTime: string;
    changeTime: string;
    birthTime: string;
    internal: {
        contentDigest: string;
        type: "Directory" | "File";
        mediaType?: string;
        description: string;
    };
}

export interface ImageSharp extends Partial<FileNode> {
    childImageSharp: {
        fluid?: ImageSharpFluid;
        fixed?: ImageSharpFixed;
    };
}
export interface ImageSharpFluid {
    aspectRatio?: number;
    sizes?: string;
    src: string;
    srcSet?: string;
    tracedSVG?: string;
    base64?: string;
}

export interface ImageSharpFixed {
    aspectRatio?: number;
    src: string;
    srcSet?: string;
    tracedSVG?: string;
    base64?: string;
    width: number;
    height: number;
}

// PAGES

export interface NodeList<T> {
    edges: Array<Edge<T>>;
}

export interface Edge<T> {
    node: T;
}
export interface PagesListData {
    data: {
        allPagesJson: PageList;
    };
}

export interface PageList {
    edges: PageListNode[];
}

export type PageListNode = Edge<PageRef>;

export interface PageRef {
    id: string;
    path: string;
    title: string;
}

export interface PageSection {
    title: string;
    image?: ImageField;
    smallImage?: ImageField;
    subtitle: string;
    blurb: string;
    activities?: Activity[];
    successStory?: SuccessStory;
    testimonial?: Testimonial;
}

export interface Page {
    title: string;
    path: string;
    heroImage?: ImageField;
    subheading: string;
    description: string;
    seoTitle: string;
    seoDescription: string;
    seoKeywords: string;
}

export interface WhatWeDo extends Page {
    sections?: PageSection[];
}

export interface MeetTheTeam extends Page {
    staff: Staff[];
}

export interface About extends Page {
    infographic?: Infographic[];
}

export type InfographicType =
    | "Single Line"
    | "Compact Single Line"
    | "Split"
    | "Bar"
    | "Vertical"
    | "Horizontal"
    | "Surround";
export interface Infographic {
    primaryText?: string;
    type: InfographicType;
    secondaryText?: string;
    image?: ImageField;
    imageCount: number;
    secondaryImage?: ImageField;
}

export interface Staff {
    name: string;
    jobTitle: string;
    live: boolean;
    imageNormal: ImageField;
    imageFunny: ImageField;
    description: string;
    fact: string;
    skills: string[];
}

export interface ReactRouterLocation {
    action: string;
    hash: string;
    key: string;
    pathname: string;
    search: string;
    state: null | string;
}

export interface Testimonial {
    quote?: string;
    person?: string;
    title?: string;
    avatar?: ImageField;
    image?: ImageField;
}

export interface Stat {
    title: string;
    text: string;
    image: ImageField;
}
export interface SuccessStory {
    title: string;
    stats?: Stat[];
}

export interface Activity {
    title: string;
    text: string;
}

export interface Hurdle {
    title: string;
    image: ImageField;
    text: string;
}

export interface ProjectStage {
    description: string;
    hurdles: Hurdle[];
}

export interface Performance {
    title: string;
    text: string;
}
export interface ProjectBox {
    name: string;
    slug: string;
    heroImage: ImageField;
    thumbnail: ImageField;
    client: Client;
    titleLineOne: string;
}
export interface Project {
    name: string;
    slug: string;
    heroImage: ImageField;
    heroVideo?: string;
    titleLineOne: string;
    titleLineTwo?: string;
    client: Client;
    goLiveDate?: string;
    caseStudyTitle: string;
    caseStudyText: string;
    caseStudyImage: ImageField;
    externalUrl: string;
    rolloverDetails: string;
    homepage: boolean;
    featured: boolean;
    gallery: GalleryImage[];
    challenge: ProjectStage;
    solution: ProjectStage;
    results: ProjectStage;
    performance?: Performance[];
    testimonial?: Testimonial;
}

export interface Client {
    name: string;
    logo: ImageField;
    promoLogo: ImageField;
    featured: boolean;
    featuredProject?: string;
    id: string;
}
