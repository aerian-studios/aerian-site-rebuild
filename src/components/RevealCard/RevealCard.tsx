import * as React from "react";

import { Project } from "../../types/data";
// import { Image } from "../Image";
import * as styles from "./RevealCard.scss";
interface Props {
    style?: React.CSSProperties;
    className?: string;
    project: Project;
}

export const RevealCard: React.SFC<Props> = ({
    children,
    style,
    className,
    project
}) => (
    <div className={[styles.component, className].join(" ")} style={style}>
        <div className="overlay">
            {/* this will have the dark transparent background */}
            <div className="content">
                <h3>{project.name}</h3>
                <a className={styles.button} href={project.slug}>
                    View more
                </a>
                {/* <Image source={project.client.promoLogo} /> */}
            </div>
        </div>
    </div>
);
export default RevealCard;
