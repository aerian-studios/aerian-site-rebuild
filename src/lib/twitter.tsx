import React from "react";

import { Tweet } from "../types/data";

export const parseTweet = (tweet: Tweet) => {
    const { urls } = tweet.entities;
    if (!urls || !urls.length) {
        return tweet.full_text;
    }

    const text = tweet.full_text;

    const elements: React.ReactNode[] = [text.substring(0, urls[0].indices[0])];
    for (let i = 0; i < urls.length; i++) {
        const url = urls[i];
        elements.push(
            <a
                key={url.url}
                href={url.expanded_url}
                target="_blank"
                rel="noreferrer noopener nofollow"
            >
                {url.display_url}
            </a>
        );
        const nextIndex =
            i === urls.length - 1 ? text.length : urls[i + 1].indices[0];
        elements.push(text.substring(url.indices[1], nextIndex));
    }

    return <>{elements}</>;
};
