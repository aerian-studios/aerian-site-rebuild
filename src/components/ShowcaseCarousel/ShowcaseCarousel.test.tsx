/* eslint-disable @typescript-eslint/no-explicit-any */
import * as React from "react";
import * as renderer from "react-test-renderer";
import { ProjectBox } from "../../types/data";
import { projectBox } from "../../types/fixtures";
import { RevealCard } from "../RevealCard";
import { ShowcaseCarousel } from "./index";
import { useStaticQuery } from "gatsby";
import { mockTweet } from "../../lib/twitter.test";

(global as any).IntersectionObserver = jest.fn(() => ({
    observe: jest.fn(),
    unobserve: jest.fn(),
    disconnect: jest.fn()
}));

const projects: ProjectBox[] = [projectBox];

describe("ShowcaseCarousel", () => {
    beforeEach(() => {
        (useStaticQuery as jest.Mock).mockReturnValue({ tweet: mockTweet });
    });
    it("renders correctly", () => {
        const tree = renderer
            .create(<ShowcaseCarousel data={projects} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders handles intersection", () => {
        const entry = {
            isIntersecting: true
        };

        const tree = renderer.create(<ShowcaseCarousel data={projects} />);

        const spy = jest.fn();
        tree.root.findByType(RevealCard).instance.setVisible = spy;
        const instance = tree.getInstance() as any;
        instance.handleChange(entry, 0);

        expect(spy).toHaveBeenCalledWith(true);
    });
});
