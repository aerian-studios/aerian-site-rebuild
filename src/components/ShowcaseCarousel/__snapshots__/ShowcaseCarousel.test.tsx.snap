// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowcaseCarousel renders correctly 1`] = `
<div
  className="  "
>
  <div
    className="fade"
  >
    <div>
      <a
        aria-label="Visit the BBC case study"
        href="/our-work/project/bbc-iwonder-perfect-diet-for-you"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className=""
        >
          <img
            alt="BBC"
            fadeIn={false}
            src="https://raw.githubusercontent.com/aerian-studios/aerian-site-rebuild/master/static/assets/imported/280/bbc-perfect-diet-for-you-hero-imagejpg.jpg"
            style={
              Object {
                "backgroundColor": "#d01944",
              }
            }
          />
        </div>
        <div
          className=""
        >
          <img
            alt="CNN's logo"
            src="https://raw.githubusercontent.com/aerian-studios/aerian-site-rebuild/master/static/assets/imported/474/cnn-white-logo.png"
          />
        </div>
        <div>
          <h3>
            CNN
          </h3>
          <p>
            What sort of 'eater' are you?
          </p>
          <button
            className="undefined"
            tabIndex={-1}
          >
            View project
            <i
              className="fa"
            />
          </button>
        </div>
      </a>
    </div>
  </div>
  <div
    className="component"
  >
    <div
      className=""
    >
      <div>
        MOCK!: proud that our partners over at @CRbuildpeace are being featured by our friends at @BBCRadio4 this Sunday. What's more, we've recently just launched their new site - 
        <a
          href="http://www.c-r.org"
          rel="noreferrer noopener nofollow"
          target="_blank"
        >
          c-r.org
        </a>
        . Check it out! 
        <a
          href="https://twitter.com/CRbuildpeace/status/1164462963790667776"
          rel="noreferrer noopener nofollow"
          target="_blank"
        >
          twitter.com/CRbuildpeace/sâ€¦
        </a>
        
      </div>
      <a
        href="https://twitter.com/aerianstudios"
      >
        <img
          alt="Aerian"
          src="https://pbs.twimg.com/profile_images/378800000156917292/cba17ed82c5a99f3b91e805534b2d10f_normal.png"
        />
        <p>
          <strong>
            Tweeted by
          </strong>
          Aerian
        </p>
      </a>
    </div>
  </div>
</div>
`;
